os/drivers/error_report/error_report_drv.c

```diff
@@ -0,0 +1,301 @@
+static ssize_t err_report_write(FAR struct file *filep, FAR const char *buffer, size_t len);
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+#define INFINITE_CHECK_HIGH_THRESHOLD	100
+static const struct file_operations err_report_fops = {
+	0,                          /* open */
+	0,                          /* close */
+	err_report_read,            /* read */
+	err_report_write,           /* write */
+	0,                          /* seek */
+	error_report_ioctl          /* ioctl */
+#ifndef CONFIG_DISABLE_POLL
+	, 0                         /* poll */
+#endif
+};
+
+static create_infwait_err_rec g_create_err_rec;
+static infinity_waitq_t g_thd_waitq[CONFIG_MAX_TASKS];
+#if CONFIG_ERROR_REPORT_INFINITE_WAIT
+static const char *g_exceptional_waits[] = {"tash", "lpwork", "hpwork", "LWIP_TCP/IP", "Network Error Reporting", "WPA Supplicant", "WLAN Driver mxmgmt", "WPA Ctrl Iface FIFO", "Wi-Fi API monitor"};
+#endif
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+#if CONFIG_ERROR_REPORT_INFINITE_WAIT
+#define INSERT_INFINITE_WAITQ(pos, tcb, cpu)        \
+		do {                                        \
+			g_thd_waitq[pos].pid = tcb->pid;        \
+			g_thd_waitq[pos].task_state = tcb->task_state;\
+			g_thd_waitq[pos].count = 1;             \
+			g_thd_waitq[pos].active_cpu = cpu.active;\
+			nwerr_vdbg("Inserted pid %d at pos %d\n", tcb->pid, pos);\
+		} while (0)
+
+#define DELETE_INFINITE_WAITQ(pos)                  \
+		do {                                        \
+			g_thd_waitq[pos].pid = 0;               \
+			g_thd_waitq[pos].count = 0;             \
+			g_thd_waitq[pos].task_state = 0;        \
+			g_thd_waitq[pos].active_cpu = 0;        \
+			nwerr_vdbg("Reset pos %d\n", pos);      \
+		} while (0)
+#endif
+
+static int prv_is_reserved(const char *taskname)
+{
+	int i;
+	if (taskname == NULL) {
+		return -1;
+	}
+	for (i = 0; i < sizeof(g_exceptional_waits)/sizeof(g_exceptional_waits[0]); i++) {
+		if (!strncmp(taskname, g_exceptional_waits[i], strlen(taskname))) {
+			return 1;
+		}
+	}
+	return 0;
+}
+
+static ssize_t err_report_read(FAR struct file *filep, FAR char *buffer, size_t len)
+{
+	return 0;
+}
+
+static ssize_t err_report_write(FAR struct file *filep, FAR const char *buffer, size_t len)
+{
+	return 0;
+}
+
+static unsigned long prv_fetch_taskaddr(int pid)
+{
+	struct tcb_s *tcbptr = sched_gettcb(pid);
+	if (tcbptr != NULL) {
+		entry_t e = tcbptr->entry;
+		if ((tcbptr->flags & TCB_FLAG_TTYPE_MASK) == TCB_FLAG_TTYPE_PTHREAD) {
+			return (unsigned long)e.pthread;
+		} else {
+			return (unsigned long)e.main;
+		}
+	}
+	return 0;
+}
+
+static void prv_tcb_handler(FAR struct tcb_s *tcb, void *args)
+{
+	int pos = 0;
+	if (tcb->pid < 5) {
```

Comment:
Why 5? for idle, hpwork, lpwork, and so on? Can you guarantee  those are always enabled?

Reply:
Removed the condition check, as these tasks are now considered as exceptional cases, and will have a larger time window for reporting purposes. Thanks again.